spec/controllers/admin/companies_controller_spec.rb

```diff
@@ -32,7 +32,17 @@
 
   describe "PATCH #update" do
     it "updates the status of a company" do
-      patch :update, id: company.id, company: { status: "Contacted" }
+      patch :update, id: company.id, company: { status: "contacted" }
+      expect(response).to redirect_to(admin_company_path(company.id))
+    end
+  end
+
+  describe "POST #edit" do
+    it "updates a company information" do
+      get :edit, id: company.id
+      expect(response).to render_template(:edit)
+
+      post :update, id: company.id, company: { name: "Netflix", status: "contacted" }
```

Comment:
I was confused here because if I don't give status here it took status to be nil and since the validation failed was not saving it in database. Let me know if anyone can find a better solution here.